comp/core/agenttelemetry/agenttelemetryimpl/agenttelemetry.go

```diff
@@ -0,0 +1,548 @@
+		a.logComp.Errorf("Failed to collect Agent Status telemetry. Error: %s", err.Error())
+		return
+	}
+}
+
+func (a *atel) addAgentStatusExtra(p *Profile, fullStatus map[string]interface{}, statusOutput map[string]interface{}) {
+	for _, builder := range p.statusExtraBuilder {
+		// Evaluate JQ expression against the agent status JSON object
+		jqResult := builder.jqSource.Run(fullStatus)
+		jqValue, ok := jqResult.Next()
+		if !ok {
+			a.logComp.Errorf("Failed to apply JQ expression for JSON path '%s' to Agent Status payload. Error unknown",
+				strings.Join(builder.jpathTarget, "."))
+			continue
+		}
+
+		// Validate JQ expression result
+		if err, ok := jqValue.(error); ok {
+			a.logComp.Errorf("Failed to apply JQ expression for JSON path '%s' to Agent Status payload. Error: %s",
+				strings.Join(builder.jpathTarget, "."), err.Error())
+			continue
+		}
+
+		// Validate JQ expression result type
+		var attrVal interface{}
+		switch jqValueType := jqValue.(type) {
+		case int:
+			attrVal = jqValueType
+		case float64:
+			attrVal = jqValueType
+		case bool:
+			attrVal = jqValueType
+		case nil:
+			a.logComp.Debugf("JQ expression return 'nil' value for JSON path '%s'", strings.Join(builder.jpathTarget, "."))
+			continue
+		case string:
+			a.logComp.Errorf("string value (%v) for JSON path '%s' for extra status atttribute is not currently allowed",
+				strings.Join(builder.jpathTarget, "."), attrVal)
+			continue
+		default:
+			a.logComp.Errorf("'%v' value (%v) for JSON path '%s' for extra status atttribute is not currently allowed",
+				reflect.TypeOf(jqValueType), reflect.ValueOf(jqValueType), strings.Join(builder.jpathTarget, "."))
+			continue
+		}
+
+		// Add resulting value to the agent status telemetry payload (recursively creating missing JSON objects)
+		curNode := statusOutput
+		for i, p := range builder.jpathTarget {
+			// last element is the attribute name
+			if i == len(builder.jpathTarget)-1 {
+				curNode[p] = attrVal
+				break
+			}
+
+			existSubNode, ok := curNode[p]
+
+			// if the node doesn't exist, create it
+			if !ok {
+				newSubNode := make(map[string]interface{})
+				curNode[p] = newSubNode
+				curNode = newSubNode
+			} else {
+				existSubNodeCasted, ok := existSubNode.(map[string]interface{})
+				if !ok {
+					a.logComp.Errorf("JSON path '%s' points to non-object element", strings.Join(builder.jpathTarget[:i], "."))
+					break
+				}
+				curNode = existSubNodeCasted
+			}
+		}
+	}
+}
+
+func (a *atel) reportAgentStatus(session *senderSession, p *Profile) {
+	// If no status is configured nothing to report
+	if p.Status == nil {
+		return
+	}
+
+	a.logComp.Debugf("Collect Agent Status telemetry for profile %s", p.Name)
+
+	// Current "agent-telemetry-basic.tmpl" uses only "runneStats" and "dogstatsdStats" JSON sections
+	// These JSON sections are populated via "collector" and "DogStatsD" status providers sections
+	minimumReqSections := []string{"collector", "DogStatsD"}
```

Comment:
nit: Since we might have more templates in the future with different required sections. We could use a map to store template and status sections information

Reply:
True, but I want to see it first.